internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -325,29 +325,36 @@ func convertAutoscaling(a manifest.AdvancedCount) (*template.AutoscalingOpts, er
 func convertHTTPHealthCheck(hc *manifest.HealthCheckArgsOrString) template.HTTPHealthCheckOpts {
 	opts := template.HTTPHealthCheckOpts{
 		HealthCheckPath:    manifest.DefaultHealthCheckPath,
-		HealthyThreshold:   hc.HealthCheckArgs.HealthyThreshold,
-		UnhealthyThreshold: hc.HealthCheckArgs.UnhealthyThreshold,
+		HealthyThreshold:   hc.B.HealthyThreshold,
+		UnhealthyThreshold: hc.B.UnhealthyThreshold,
 		GracePeriod:        manifest.DefaultHealthCheckGracePeriod,
 	}
-	if hc.HealthCheckArgs.Path != nil {
-		opts.HealthCheckPath = *hc.HealthCheckArgs.Path
-	} else if hc.HealthCheckPath != nil {
-		opts.HealthCheckPath = *hc.HealthCheckPath
+
+	if hc.IsZero() {
+		return opts
+	}
+	if hc.IsA() {
+		opts.HealthCheckPath = hc.A
```

Comment:
I wonder in Copilot since all the union type consists of a simpler one and a more complicated one: like `string` and some struct, maybe we can make `A` and `B` more informative.

Reply:
do you think it would make sense in all/enough cases to change `A` -> `Simple` and `B` -> `Complex`, or something like that? that could make it a little better for reading 🤔

Reply:
Yeah i think simple and complex sounds good enough to me.

Reply:
I just checked all of our structs - pretty much all are `[builtinType, struct]`, but a few are a bit more complex:
```
SubnetListOrArgs -> []string | SubnetArgs
SecurityGroupsIDsOrConfig -> []string | SecurityGroupsConfig
```
I'll add a test to confirm those types work (I may already have a test, just want to make sure :)) I think I'm still ok with [`Simple`/`Advanced`](https://github.com/aws/copilot-cli/pull/4100#discussion_r1001983070) in those scenarios, what do you think?

Reply:
:shipit:

Reply:
Ooh I like "advanced" more than "complex"...how do you feel about "basic" over "simple"?

Reply:
i don't care _too_ much either way - i feel like Basic sorta implies a subset (but still a struct) of Advanced though? as opposed to a single field? maybe that's just me though 🙄

Reply:
i changed the name to Basic😊. It _does_ support `[structA, structB]` so I think that it makes more sense that way